cms/static/js/models/settings/course_details.js

```diff
@@ -60,8 +60,8 @@ define(['backbone', 'underscore', 'gettext', 'js/models/validation_helpers', 'js
                 if (newattrs.end_date && newattrs.enrollment_end && newattrs.end_date < newattrs.enrollment_end) {
                     errors.enrollment_end = gettext('The enrollment end date cannot be after the course end date.');
                 }
-                if (!this.get('self_paced') && newattrs.end_date && newattrs.certificate_available_date &&
-                    newattrs.certificate_available_date < newattrs.end_date) {
+                if (this.show_certificate_available_date && !this.get('self_paced') && newattrs.end_date &&
```

Comment:
You can get rid of the self_paced clause here - `can_show_certificate_available_date_field()` checks that attribute of the course.
(You'll also want to update to camel case here.)